src/main/java/com/google/devtools/build/lib/bazel/bzlmod/VendorManager.java

```diff
@@ -83,14 +89,56 @@ public void vendorRepos(Path externalRepoRoot, ImmutableList<RepositoryName> rep
       // 3. Move the external repo to vendor dir. It's fine if this step fails or is interrupted,
       // because the marker file under external is gone anyway.
       FileSystemUtils.moveTreesBelow(repoUnderExternal, repoUnderVendor);
-      // 4. Rename to temporary marker file after the move is done.
+      // 4. Re-plant symlinks pointing a path under the external root to a relative path
+      // to make sure the vendor src keep working after being moved or output base changed
+      replantSymlinks(repoUnderVendor, externalRepoRoot);
+      // 5. Rename to temporary marker file after the move is done.
       tMarker.renameTo(markerUnderVendor);
-      // 5. Leave a symlink in external dir.
+      // 6. Leave a symlink in external dir.
       repoUnderExternal.deleteTree();
       FileSystemUtils.ensureSymbolicLink(repoUnderExternal, repoUnderVendor);
     }
   }
 
+  /**
+   * Replants the symlinks under the specified repository directory.
+   *
+   * Re-write symlinks that originally pointing to a path under the external root to a relative path pointing
+   * to an external root symlink under the vendor directory.
+   *
+   * @param repoUnderVendor The path to the repository directory under the vendor directory.
+   * @param externalRepoRoot The path to the root of external repositories.
+   * @throws IOException If an I/O error occurs while replanting the symlinks.
+   */
+  private void replantSymlinks(Path repoUnderVendor, Path externalRepoRoot) throws IOException {
+    try {
+      Collection<Path> symlinks = FileSystemUtils.traverseTree(repoUnderVendor, Path::isSymbolicLink);
```

Comment:
Could we inline this into the `moveTree` file system traversal to avoid another one?

Reply:
Hmm, I have to re-implement the moveTreesBelow to inline this. I'm not so convinced of the trade off between the performance gain and the duplicated code. Do you really think this optimization very important?

Reply:
Not at this point, no. But maybe we could add some profiler spans?

Reply:
Good point! Profile added.